types/params.go

```diff
@@ -157,7 +145,7 @@ func (params ConsensusParams) Update(params2 *abci.ConsensusParams) ConsensusPar
 		res.Evidence.MaxAge = params2.Evidence.MaxAge
 	}
 	if params2.Validator != nil {
-		res.Validator.PubKeyTypes = params2.Validator.PubKeyTypes
+		res.Validator.PubKeyTypes = append(params2.Validator.PubKeyTypes[0:0], params2.Validator.PubKeyTypes...)
```

Comment:
good point! I was trying to avoid an allocation which sets everything to 0's and then a write. I'll add a comment that this is efficiently copying it.

Reply:
why aren't we just using `copy` ? What effect does this even have? Isn't it copying right back into the same location ?